src/coprocessor/codec/mysql/json/binary.rs

```diff
@@ -215,12 +218,12 @@ pub trait JsonEncoder: NumberEncoder {
     }
 }
 
-impl<T: Write> JsonEncoder for T {}
+impl<T: NumberEncoder> JsonEncoder for T {}
 
 impl Json {
```

Comment:
Prefer to be a JsonDecoder

Reply:
the JsonDecoder still use `read_slice` in some place ,  any suggestion to make `read_slice` adapt to `BufferReader`?